app/models/ems_event.rb

```diff
@@ -146,6 +146,34 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
```

Comment:
I'd like to find a way to get this block into the RHEV provider codebase somehow.  At first I only noticed the RHEV-specific events, but then I just saw the `Redhat` namespaced classes below.

AFAIK, we don't have a mechanism today for scoping this `process_vm_in_event` method for provider-specific logic.  So, it's possible we will just let this slide and mark it as something to refactor.

But, @durandom can you give a quick peek and see if you have any ideas here?